lib/taskana-core/src/test/java/acceptance/task/UpdateTaskCommentAccTest.java

```diff
@@ -1,109 +1,149 @@
+  @WithAccessId(user = "businessadmin")
   @Test
   void should_UpdateTaskComment_For_TaskComment() throws Exception {
-
-    TaskService taskService = taskanaEngine.getTaskService();
-
-    List<TaskComment> taskComments =
-        taskService.getTaskComments("TKI:000000000000000000000000000000000025");
-    assertThat(taskComments).hasSize(1);
-    assertThat(taskComments.get(0).getTextField()).isEqualTo("some text in textfield");
-
+    Task task = createDefaultTask().buildAndStore(taskService);
     TaskComment taskComment =
-        taskService.getTaskComment("TCI:000000000000000000000000000000000003");
-    taskComment.setTextField("updated textfield");
+        TaskCommentBuilder.newTaskComment()
+            .taskId(task.getId())
+            .textField("some text in textfield")
+            .buildAndStore(taskService);
 
+    taskComment.setTextField("updated textfield");
     taskService.updateTaskComment(taskComment);
 
-    List<TaskComment> taskCommentsAfterUpdate =
-        taskService.getTaskComments("TKI:000000000000000000000000000000000025");
-    assertThat(taskCommentsAfterUpdate.get(0).getTextField()).isEqualTo("updated textfield");
+    List<TaskComment> taskCommentsAfterUpdate = taskService.getTaskComments(task.getId());
+    assertThat(taskCommentsAfterUpdate)
+        .extracting(TaskComment::getTextField)
+        .containsExactly("updated textfield");
   }
 
   @WithAccessId(user = "user-1-1")
   @Test
   void should_FailToUpdateTaskComment_When_UserHasNoAuthorization() throws Exception {
-
-    TaskService taskService = taskanaEngine.getTaskService();
-
-    List<TaskComment> taskComments =
-        taskService.getTaskComments("TKI:000000000000000000000000000000000000");
-    assertThat(taskComments).hasSize(3);
-    assertThat(taskComments.get(0).getTextField()).isEqualTo("some text in textfield");
-
+    Task task = createDefaultTask().buildAndStore(taskService, "businessadmin");
     TaskComment taskComment =
-        taskService.getTaskComment("TCI:000000000000000000000000000000000001");
+        TaskCommentBuilder.newTaskComment()
+            .taskId(task.getId())
+            .textField("some text in textfield")
+            .buildAndStore(taskService, "businessadmin");
     taskComment.setTextField("updated textfield");
 
     assertThatThrownBy(() -> taskService.updateTaskComment(taskComment))
         .isInstanceOf(NotAuthorizedException.class);
 
     // make sure the task comment wasn't updated
     List<TaskComment> taskCommentsAfterUpdateAttempt =
-        taskService.getTaskComments("TKI:000000000000000000000000000000000000");
-    assertThat(taskCommentsAfterUpdateAttempt.get(0).getTextField())
-        .isEqualTo("some text in textfield");
+        taskanaEngine.runAsAdmin(wrap(() -> taskService.getTaskComments(task.getId())));
+    assertThat(taskCommentsAfterUpdateAttempt)
+        .extracting(TaskComment::getTextField)
+        .containsExactly("some text in textfield");
```

Comment:
@gitgoodjhe I'd love a short statement here. Do we really have to verify that the TaskComment has not been modified when our API throws an exception?

This is the only test where this addition is done. What do you think? Keep this, throw this away or add it to every other test aswell?

Reply:
Legit question. I think we should trust in our API here and throw this away